packages/flutter/lib/src/animation/animations.dart

```diff
@@ -533,3 +533,66 @@ class TrainHoppingAnimation extends Animation<double>
     return '$currentTrain\u27A9$runtimeType(no next)';
   }
 }
+
+/// An interface for combining multiple Animations. Subclasses need only
+/// implement the `value` getter to control how the child animations are
+/// combined. Can be chained to combine more than 2 animations.
+/// Example: to create an animation that is the sum of two others, subclass
+/// this class and define value = first.value + second.value;
+abstract class CompoundAnimation<T> extends Animation<T>
+  with AnimationLazyListenerMixin, AnimationLocalListenersMixin, AnimationLocalStatusListenersMixin {
+  /// Creates a CompoundAnimation. Both arguments must be non-null. Either can
+  /// be a CompoundAnimation itself to combine multiple animations.
+  CompoundAnimation({
+    this.first,
+    this.next,
+  }) {
+    assert(first != null);
+    assert(next != null);
+  }
+
+  /// The first sub-animation. Its status takes precedence if neither are
+  /// animating.
+  final Animation<T> first;
+
+  /// The second sub-animation.
+  final Animation<T> next;
+
+  @override
+  void didStartListening() {
+    first.addListener(notifyListeners);
+    first.addStatusListener(_maybeNotifyStatusListeners);
+    next.addListener(notifyListeners);
+    next.addStatusListener(_maybeNotifyStatusListeners);
+  }
+
+  @override
+  void didStopListening() {
+    first.removeListener(notifyListeners);
+    first.removeStatusListener(_maybeNotifyStatusListeners);
+    next.removeListener(notifyListeners);
+    next.removeStatusListener(_maybeNotifyStatusListeners);
+  }
+
+  @override
+  AnimationStatus get status {
+    // If one of the sub-animations is moving, use that status. Otherwise,
+    // default to `first`.
+    if (next.status == AnimationStatus.forward || next.status == AnimationStatus.reverse)
+      return next.status;
+    return first.status;
+  }
+
+  @override
+  String toString() {
+    return '$runtimeType($first, $next)';
+  }
+
+  AnimationStatus _lastStatus;
+  void _maybeNotifyStatusListeners(AnimationStatus _) {
+    if (status != _lastStatus) {
+      _lastStatus = status;
+      notifyStatusListeners(status);
+    }
+  }
+}
```

Comment:
Sure. I'm saying, check what value you would return, and only fire the notifications if the value itself changed. That way when both _are_ animating, you only fire one notification here, rather than two, with the second not having actually changed.